src/components/UsersDetailsCard.jsx

```diff
@@ -107,61 +122,66 @@ export default function UsersDetailsCard() {
   };
 
   const selectedIdArray = (id) => {
-    let arr = [...selectedArray];
+    let arr = [...selectedUsersIdArray];
     if (arr.includes(id)) {
-      setSelectedArray(arr.filter((ele) => ele !== id));
-      setSelectAll(false);
+      setState({ selectedUsersIdArray: arr.filter((ele) => ele !== id) });
+      setState({ selectAll: false });
     } else {
       arr.push(id);
-      setSelectedArray(arr);
-      if (arr.length === usersData.length) setSelectAll(true);
-      else setSelectAll(false);
+      setState({ selectedUsersIdArray: arr });
+      if (arr.length === usersData.length) setState({ selectAll: true });
```

Comment:
Why extra if else, they can be a single statement.
```
setState({ selectAll: (arr.length === usersData.length) });
```

Reply:
done